### ISSUE TITLE
Components Manager

### ISSUE THREAD

**Issue by silverbackdan** [components-web-app/cwa-nuxt-module#34]
Tasks to complete the first edition of features for the Components Manager bar.

- [x] Implement vue-multiselect input with usage for styles by component tab
- [x] Order manager tab including the required new input types (buttons, toggle)
- [x] Status / Publishable tab - many features to implement. Refer to api tests for this feature
- [x] Ensure the implementation has been simple enough by configuring our dummy html component to have its own tab with a toggle to trigger the Quill HTML editor. This may require an integrated event system to make it easy for a tab to emit an event and a component to receive it.
- [ ] Buttons in the manager to reuse component or add component if new - a configurable system for showing buttons where required
- [x] Add a component - deal with how we are saving a dummy component data set and submitting it after. Then how we change this dummy component to the newly created one that'll be returned. Updating the component collection... can we re-order the component before it's added so it adds in the right place, can we prevent the adding process from being cancelled without first having a warning message so a user does not lose data... Adding a component maybe should be its own issue.

Once the above is done, then the components manager is ready in its simplest form. We may need to look into the paths and the overflow when they are long. How we manage page routes as well.

The manager will need reviewing in how it is used on a new dynamic page.

**Comment by silverbackdan** [comment]
Closing and reopening more targeted issues
